language/move-stdlib/src/natives/debug.rs

```diff
@@ -483,6 +483,26 @@ mod testing {
                                 ));
                         }
 
+                        let str = move_value_as_escaped_string(val)?;
+                        write!(out, "\"{}\"", str).map_err(fmt_error_to_partial_vm_error)?
+                    } else if !canonicalize && type_.is_ascii_string(move_std_addr) {
```

Comment:
Does it not work to combine these two cases? Something like `if !canonicalize && (type_.is_std_string(move_std_addr) || type_.is_ascii_string(move_std_addr))`

Reply:
Code looks _really_ similar, but couldn't tell if it would work or not from staring at it

Reply:
It would work, but it feels fragile (e.g., relies on the fact that both just happen to have a field named `bytes`).